pkg/controller/v1beta1/inferenceservice/controller.go

```diff
@@ -359,6 +370,21 @@ func (r *InferenceServiceReconciler) SetupWithManager(mgr ctrl.Manager, deployCo
 		r.Log.Info("The InferenceService controller won't watch networking.istio.io/v1beta1/VirtualService resources because the CRD is not available.")
 	}
 
+	if ingressConfig.EnableGatewayAPI {
+		gatewayapiFound, err := utils.IsCrdAvailable(r.ClientConfig, gatewayapiv1.GroupVersion.String(), constants.HTTPRouteKind)
+		if err != nil {
+			return err
+		}
+
+		if gatewayapiFound {
+			ctrlBuilder = ctrlBuilder.Owns(&gatewayapiv1.HTTPRoute{})
+		} else {
+			r.Log.Info("The InferenceService controller won't watch gateway.networking.k8s.io/v1/HTTPRoute resources because the CRD is not available.")
```

Comment:
probably should panic here if gateway API is enabled but gateway CRD is not installed

Reply:
Done